File: src/main/java/um/haberes/rest/model/CodigoImputacion.java
/**
 * 
 */
package um.haberes.rest.model;

import java.io.Serializable;
import java.math.BigDecimal;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import jakarta.persistence.UniqueConstraint;

import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 * @author daniel
 *
 */
@Entity
@Table(uniqueConstraints = {
		@UniqueConstraint(columnNames = { "dependenciaId", "facultadId", "geograficaId", "codigoId" }) })
@EqualsAndHashCode(callSuper = false)
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class CodigoImputacion extends Auditable implements Serializable {
	/**
	* 
	*/
	private static final long serialVersionUID = -24094967294542981L;

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long codigoImputacionId;

	private Integer dependenciaId;
	private Integer facultadId;
	private Integer geograficaId;
	private Integer codigoId;
	private BigDecimal cuentaSueldosDocente;
	private BigDecimal cuentaAportesDocente;
	private BigDecimal cuentaSueldosNoDocente;
	private BigDecimal cuentaAportesNoDocente;

	public String key() {
		return dependenciaId + "." + facultadId + "." + geograficaId + "." + codigoId;
	}

}

File: src/main/java/um/haberes/rest/kotlin/model/CodigoImputacion.kt
package um.haberes.rest.kotlin.model

import jakarta.persistence.*
import um.haberes.rest.model.Auditable
import java.math.BigDecimal

@Entity
@Table(uniqueConstraints = [UniqueConstraint(columnNames = ["dependenciaId", "facultadId", "geograficaId", "codigoId"])])
data class CodigoImputacion(

    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    var codigoImputacionId: Long? = null,

    var dependenciaId: Int? = null,
    var facultadId: Int? = null,
    var geograficaId: Int? = null,
    var codigoId: Int? = null,
    var cuentaSueldosDocente: BigDecimal? = null,
    var cuentaAportesDocente: BigDecimal? = null,
    var cuentaSueldosNoDocente: BigDecimal? = null,
    var cuentaAportesNoDocente: BigDecimal? = null


) : Auditable() {
    fun key(): String {
        return dependenciaId.toString() + "." + facultadId + "." + geograficaId + "." + codigoId
    }

}
